src/environments/ui.ts

```diff
@@ -0,0 +1,462 @@
+/* eslint-disable */
+import { environment } from './environment';
+import { environmentTemplate } from './environment.template';
+import { Command } from 'commander';
+import * as fs from 'fs';
+import {EnclosureDispersalStrategy} from "../app/core/testing/enums/mock-storage.enum";
+const figlet = require("figlet");
+import { WebUiEnvironment } from "./environment.interface";
+
+interface CommandOptions {
+  [p: string]: any;
+}
+
+interface ReportOptions {
+  data?: unknown;
+  showHeader: boolean;
+  showFooter: boolean;
+}
+
+interface Headers {
+  header: string;
+  footer: string;
+}
+
+
+/*
+* Nice Header
+* */
+function showBanner(): void {
+  console.log(figlet.textSync('TrueNAS WebUI'));
+}
+
+function generateHeaders(content: string): Headers {
+  const width: number = 44;
+  const start = '\n ';
+  const finish = '\n';
+  const asterisk = '*';
+  const asterisksPerSide = (width - (content.length + 2)) / 2;
+  const asterisks = asterisk.repeat(asterisksPerSide);
+  const output = {
+    header: start +asterisks + ' ' + content.toUpperCase() + ' ' + asterisks + finish,
+    footer: start +asterisk.repeat(width) + '\n\n',
+  }
+
+  return output;
+}
+
+/*
+* Variables
+* */
+const environmentTs = './src/environments/environment.ts';
+const template = './src/environments/environment.template.ts';
+const originalEnvironment = {...environment};
+
+/*
+* Command Setup
+* */
+const program: Command = new Command();
+program
+  .version('0.0.1')
+  .description('WebUI setup utility')
+
+program
+  .command('mock')
+  .description('Mocking enclosure ui support')
+  .option('-r, --reset', 'Reset mock configuration option')
+  .option('-D, --debug', 'debug output')
+  .option('-c, --config <filename>', 'load mock config file')
+  .option('-e, --enable', 'enable mock config')
+  .option('-d, --disable', 'disable mock config')
+  .option('-m, --model <name>', 'set controller model to mock')
+  .option('-M, --showcontrollers ', 'show available controllers for mock')
+  .option('-l, --list', 'show current mock config settings')
+  .option('-a, --assign, <existing | default>', 'set slot assignment strategy')
+  .option('-s, --shelves, <list as string>', 'set expansion shelves')
+  .option('-S, --showshelves ', 'show available shelves for mock')
+  .action(() => {
+    const mockCommand = program.commands.find((command: Command) => command._name === 'mock');
+    const mockOptions = commandOpts(mockCommand);
+    mock(mockCommand, mockOptions);
+  });
+
+program
+  .command('reset')
+  .description('Reset environment file to initial state')
+  .action(() => {
+    reset();
+    process.exit(0);
+    console.log(environment);
+
+    showRemote({
+      showHeader: true,
+      showFooter: false,
+    });
+    mockConfigReport({
+      showHeader: true,
+      showFooter: true,
+    });
+  });
+
+program
+  .command('remote')
+  .description('Set the server WebUI communicates with')
+  .option('-i, --ip <ip_address>', 'Sets IP address of your server')
```

Comment:
Probably don't need a flag.

Reply:
Just for future reference. As per our discussions, the design was to have the command without flags get the current state.  Flags are what we use to set values.